lib/wallet/wallet.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:ten_ten_one/menu.dart';
+import 'package:ten_ten_one/wallet/wallet_bitcoin.dart';
+import 'package:ten_ten_one/wallet/wallet_change_notifier.dart';
+import 'package:ten_ten_one/wallet/wallet_dashboard.dart';
+import 'package:ten_ten_one/wallet/wallet_lightning.dart';
+
+class Wallet extends StatefulWidget {
+  static const route = '/cfd-trading';
+
+  const Wallet({Key? key}) : super(key: key);
+
+  @override
+  State<Wallet> createState() => _WalletState();
+}
+
+class _WalletState extends State<Wallet> {
+  final List<Widget> _pages = <Widget>[
+    const WalletDashboard(),
+    const WalletLightning(),
+    const WalletBitcoin(),
+  ];
+
+  final List<String> _pageNames = <String>[
+    WalletDashboard.name,
+    WalletLightning.name,
+    WalletBitcoin.name,
+  ];
+
+  final List<String> _pageNavigationLabels = <String>[
+    WalletDashboard.navigationLabel,
+    WalletLightning.navigationLabel,
+    WalletBitcoin.navigationLabel,
+  ];
+
+  final List<IconData> _pageIcons = <IconData>[
+    WalletDashboard.icon,
+    WalletLightning.icon,
+    WalletBitcoin.icon,
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    WalletChangeNotifier walletChangeNotifier = context.watch<WalletChangeNotifier>();
+
+    return Scaffold(
+        appBar: AppBar(
+            // title: walletChangeNotifier.selectedIndex == 0 ? const Text('Lightning Wallet') : const Text('Bitcoin Wallet'),
+            title: Text(_pageNames.elementAt(walletChangeNotifier.selectedIndex))),
+        drawer: const Menu(),
+        body: _pages.elementAt(walletChangeNotifier.selectedIndex),
+        bottomNavigationBar: BottomNavigationBar(
+          items: List.generate(
```

Comment:
❓ why extracting the list to the top. It seems unnecessary complicated without much benefits. I suggest writing the bottom navigation bar items directly here.

Alternatively if you insist on that structure, I would at least create a new class wrapping label and icon and create a list from them. Then you can write something like this.

```
final List<Item> navigationBarItems = [...]
...
navigationBarItems.map((item) => BottomNavigationBarItem(icon: item.icon, label: item.label)).toList()
```

Reply:
feel free to create follow up for this :)